vernac/vernacentries.ml

```diff
@@ -510,11 +510,15 @@ let vernac_open_close_scope ~section_local (b,s) =
 let vernac_arguments_scope ~section_local r scl =
   Notation.declare_arguments_scope section_local (smart_global r) scl
 
-let vernac_infix ~module_local =
-  Metasyntax.add_infix module_local (Global.env())
+let vernac_infix ~atts =
+  let local, deprecation = Attributes.(parse Notations.(locality ++ deprecation) atts) in
+  let module_local = make_module_locality local in
```

Comment:
Would it make sense to define a `module_locality = locality >>= fun l -> return (make_module_locality l)` attribute?